static/src/javascripts/projects/commercial/modules/messenger/post-message.ts

```diff
@@ -1,7 +1,7 @@
 export const postMessage = (
 	message: Record<string, unknown>,
-	targetWindow: WindowProxy,
-	targetOrigin?: string | null | undefined,
+	target: MessageEventSource,
+	targetOrigin = '*',
 ): void => {
-	targetWindow.postMessage(JSON.stringify(message), targetOrigin ?? '*');
+	target.postMessage(JSON.stringify(message), { targetOrigin });
```

Comment:
Are you sure the options should be passed as an object? [MDN suggest otherwise](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#syntax).

Reply:
It looks like [its overloaded!](https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L17278-L17279)

Reply:
You might want to [update the MDN docs, then](https://github.com/mdn/content/edit/main/files/en-us/web/api/window/postmessage/index.md)!

Also, I would point to the [W3C spec](https://html.spec.whatwg.org/multipage/web-messaging.html#dom-window-postmessage-options-dev) rather than its TypeScript implementation 😉.